contracts/src/matching/HyperdriveMatchingEngine.sol

```diff
@@ -0,0 +1,472 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.22;
+
+import { IMorpho } from "morpho-blue/src/interfaces/IMorpho.sol";
+import { IERC1271 } from "openzeppelin/interfaces/IERC1271.sol";
+import { ERC20 } from "openzeppelin/token/ERC20/ERC20.sol";
+import { SafeERC20 } from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import { ECDSA } from "openzeppelin/utils/cryptography/ECDSA.sol";
+import { EIP712 } from "openzeppelin/utils/cryptography/EIP712.sol";
+import { ReentrancyGuard } from "openzeppelin/utils/ReentrancyGuard.sol";
+import { IHyperdrive } from "../interfaces/IHyperdrive.sol";
+import { IHyperdriveMatchingEngine } from "../interfaces/IHyperdriveMatchingEngine.sol";
+import { AssetId } from "../libraries/AssetId.sol";
+import { HYPERDRIVE_MATCHING_ENGINE_KIND, VERSION } from "../libraries/Constants.sol";
+import { FixedPointMath } from "../libraries/FixedPointMath.sol";
+
+// TODO: Document the simplifications that were made.
```

Comment:
Not sure if you meant to do this in this PR

Reply:
I just left this here as a reminder to include the simplifications in the PR description